src/PointwiseFunctions/GeneralRelativity/WeylPropagating.cpp

```diff
@@ -0,0 +1,130 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/GeneralRelativity/WeylPropagating.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "PointwiseFunctions/GeneralRelativity/WeylElectric.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+/// \cond
+namespace gr {
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::ii<DataType, SpatialDim, Frame> weyl_propagating(
+    const tnsr::ii<DataType, SpatialDim, Frame>& ricci,
+    const tnsr::ii<DataType, SpatialDim, Frame>& extrinsic_curvature,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const tnsr::ijj<DataType, SpatialDim, Frame>& cov_deriv_extrinsic_curvature,
+    const tnsr::I<DataType, SpatialDim, Frame>& unit_interface_normal_vector,
+    const tnsr::II<DataType, SpatialDim, Frame>& projection_IJ,
+    const tnsr::ii<DataType, SpatialDim, Frame>& projection_ij,
+    const tnsr::Ij<DataType, SpatialDim, Frame>& projection_Ij,
+    const short sign) noexcept {
+  tnsr::ii<DataType, SpatialDim, Frame> weyl_prop{};
+  weyl_propagating<SpatialDim, Frame, DataType>(
+      make_not_null(&weyl_prop), ricci, extrinsic_curvature,
+      inverse_spatial_metric, cov_deriv_extrinsic_curvature,
+      unit_interface_normal_vector, projection_IJ, projection_ij, projection_Ij,
+      sign);
+  return weyl_prop;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void weyl_propagating(
+    const gsl::not_null<tnsr::ii<DataType, SpatialDim, Frame>*> weyl_prop_u8,
+    const tnsr::ii<DataType, SpatialDim, Frame>& ricci,
+    const tnsr::ii<DataType, SpatialDim, Frame>& extrinsic_curvature,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const tnsr::ijj<DataType, SpatialDim, Frame>& cov_deriv_extrinsic_curvature,
+    const tnsr::I<DataType, SpatialDim, Frame>& unit_interface_normal_vector,
+    const tnsr::II<DataType, SpatialDim, Frame>& projection_IJ,
+    const tnsr::ii<DataType, SpatialDim, Frame>& projection_ij,
+    const tnsr::Ij<DataType, SpatialDim, Frame>& projection_Ij,
+    const short sign) noexcept {
+  if (UNLIKELY(get_size(get<0, 0>(*weyl_prop_u8)) !=
```

Comment:
can you use `destructive_resize_components`?

Reply:
done